TakingITGlobal/commit2act
# [TIG-737] User preferences model and fetch requests

## Ticket(s)

* [TIG-737](https://app.clickup.com/t/9009201449/TIG-737) - focused child task
* [TIG-732](https://app.clickup.com/t/9009201449/TIG-732) - parent task

## Description

~~**Note: Do not merge this PR until [TIG-736](https://github.com/TakingITGlobal/commit2act/pull/117) is reviewed and merged!**~~

This PR adds mocked data request hooks for updating the user profile and preferences entries, as well as an initial `useGetCurrentUserPreferences` hook to pull from a (mocked) existing user preferences record. 

There will be some updates and changes to this work following [our direction change on preferences as a dedicated DB table](https://heynova.slack.com/archives/C03PWL6HPM1/p1711050316605269). This should be done after this PR has been merged and is available with the other related work to make wider changes in one go. 

## Testing

1. Pull down and install the branch (`npm install`)
2. Run all tests (`npm run test:basic`)
3. Run the app locally (`npm run start:dev`)
4. Login with a test user
5. Navigate to `/onboard/step-4`
6. Ensure each radio option for subject familiarity can be toggled by keyboard or by clicking the radio button or label
7. Hit the "NEXT" button
8. Once the "Loading..." text has passed, confirm you are redirected to `/onboard/step-5`
9. Confirm your selected option appears in the `Response` output in the browser console
10. Shut down the local app and run Storybook (`npm run storybook`)

## Screenshots

**Familiarity page with radios for level of subject knowledge:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/73c526c2-ee5f-4e34-a458-8d22a5d7afaf)

**Factoid page rendered with selected value in the console output:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/e5267c5c-222b-4f37-9243-7b0f1d42bde7)
